File: code_examples/3-Structured_Problem_Solving/9-Select_Case/JavaScript/main.js
/* Created by: Mr. Coxall
 * Created on: Sep 2020
 * This program checks a student's grade
 */

const prompt = require("prompt-sync")()

// input
const grade = prompt("Enter your grade: ")

// process & output
switch (grade.toUpperCase()) {
  case "A":
    console.log("Excellent!")
    break
  case "B":
    console.log("Good job!")
    break
  case "C":
    console.log("Average.")
    break
  case "D":
    console.log("Poor.")
    break
  case "F":
    console.log("Fail.")
    break
  default:
    console.log("Invalid grade.")
    break
}

console.log("\nDone.")

File: code_examples/3-Structured_Problem_Solving/9-Select_Case/Go/main.go
/**
 * Created by: Mr. Coxall
 * Created on: Sep 2020
 * This program checks a student's grade
 */

package main

import (
	"fmt"
	"strings"
)

func main() {
	// this function checks a student's grade

	var grade string

	// input
	fmt.Print("Enter your grade: ")
	fmt.Scanln(&grade)

	// process and output
	switch strings.ToUpper(grade) {
		case "A":
			fmt.Println("Excellent!")
		case "B":
			fmt.Println("Good job!")
		case "C":
			fmt.Println("Average.")
		case "D":
			fmt.Println("Poor.")
		case "F":
			fmt.Println("Fail.")
		default:
			fmt.Println("Invalid grade.")
	}

	fmt.Println("\nDone.")
}
